typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,151 @@
+import { ServiceID } from './service';
+import { StaticFileID } from './container_creation_config'; 
+import { newGetExecCommandArgs, newGetGenerateFilesArgs, newGetFileGenerationOptions, newGetLoadStaticFilesArgs } from "../constructor_calls";
+import * as grpc from "grpc";
+import * as path from "path";
+
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class GeneratedFileFilepaths {
+	
+    private readonly absoluteFilepathHere: string;
+	private readonly absoluteFilepathOnServiceContainer: string;
+
+    constructor (
+        absoluteFilepath: string,
+        absoluteFilepathOnServiceContainer: string) {
+            this.absoluteFilepathHere = absoluteFilepath;
+            this.absoluteFilepathOnServiceContainer = absoluteFilepathOnServiceContainer;
+    }   
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getAbsoluteFilepathHere(): string { 
+        return this.absoluteFilepathHere;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getAbsoluteFilepathOnServiceContainer(): string { 
+        return this.absoluteFilepathOnServiceContainer;
+    }
+}
+
+
+
+class ServiceContext {
+    
+    private readonly client: ApiContainerServiceClient;
+    private readonly serviceId: ServiceID;
+    private readonly ipAddress: string;
+    private readonly enclaveDataVolMountpointHere: string;
+    private readonly enclaveDataVolMountpointOnServiceContainer: string;
+
+    constructor(
+        client: ApiContainerServiceClient,
+        serviceId: ServiceID,
+        ipAddress: string,
+        enclaveDataVolMountpointHere: string,
+        enclaveDataVolMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.enclaveDataVolMountpointHere = enclaveDataVolMountpointHere;
+            this.enclaveDataVolMountpointOnServiceContainer = enclaveDataVolMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceID(): ServiceID { 
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getIPAddress(): string {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public execCommand(command: string[]): [number, Uint8Array | string, Error] {
+        const serviceId: ServiceID = this.serviceId;
+        const args: ExecCommandArgs = newGetExecCommandArgs(serviceId, command);
+
+        var resp: ExecCommandResponse;
+        var error: grpc.ServiceError;
+        this.client.execCommand(args, function(error, feature){ //TODO (comment) - don't use return value, but use callback parameters
```

Comment:
ooo I _think_ this guy won't actually do what you want. What I _think_ will happen:
1. this `client.execCommand` call will run
1. the rest of the code will proceed, with `if (error != null) {`, and all the statements following it
1. at some point in the future, when the `execCommand` call actually comes back, the callback will run and `error` and `resp` will be set (but by then, far too late).

However:  let's not fuss about this for right now; go ahead and keep using this style, but just be mindful that when it comes time to running this probably won't work.